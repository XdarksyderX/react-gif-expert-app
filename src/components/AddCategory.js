import React, { useState } from 'react';
import PropTypes from 'prop-types';
export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		if (inputValue.trim().length > 2) {
			setCategories((prev) => [inputValue, ...prev]);
			setInputValue('');
		}
	};

	return (
		<form onSubmit={handleSubmit}>
			<input type="text" value={inputValue} onChange={handleInputChange} />
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
