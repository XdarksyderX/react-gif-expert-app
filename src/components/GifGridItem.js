import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
	return (
		<div className="card animate__animated animate__fadeIn animate__delay-1s">
			<img src={url} alt={title} />
			<p> {title} </p>
		</div>
	);
};

GifGridItem.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired
}