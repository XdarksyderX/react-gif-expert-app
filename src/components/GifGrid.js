import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);
	return (
		<div className="gif-grid animate__animated animate__bounceInDown">
			<h3>{category}</h3>
			{loading ? <p className="animate__animated animate__fadeIn animate__infinite">Loading...</p> : null}
			<ul className="gifs-container">
				{images.map((img) => {
					return <GifGridItem key={img.id} url={img.url} title={img.title} />;
				})}
			</ul>
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};
