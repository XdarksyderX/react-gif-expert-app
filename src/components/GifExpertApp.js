import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(['One Punch']);

	return (
		<div className="app">
			<h2>GifExpertApp</h2>
			
			<AddCategory setCategories={setCategories}/>
            <ul>
                {categories.map(cat => <GifGrid key={cat} category={cat}/>)}
            </ul>
			<hr />
		</div>
	);
};
