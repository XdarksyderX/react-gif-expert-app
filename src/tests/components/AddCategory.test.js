import React from 'react';
import { shallow } from 'enzyme';

import { AddCategory } from '../../components/AddCategory';

const setCategories = jest.fn();

describe('AddCategory', () => {
	let wrapper = shallow(<AddCategory setCategories={setCategories} />);

	beforeEach(() => {
		jest.clearAllMocks();
	});

	test('should display correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should change the text box', () => {
		const input = wrapper.find('input');
        const value = '';

		input.simulate('change', {
			target: {
				value,
			},
		});
	});

    test('shouldn`t post the information ', () => {
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call setcategories and clean the inputValue box', () => {
        const input = wrapper.find('input');
        input.simulate('change', {
            target: {
                value: 'Hola Mundo'
            }
        })
        wrapper.find('form').simulate('submit', {preventDefault(){}});
        expect(setCategories).toHaveBeenCalled();
        expect(input.prop('value')).toBe('');
    });

    
});
