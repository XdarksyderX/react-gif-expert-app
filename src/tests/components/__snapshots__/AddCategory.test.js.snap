// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory should display correctly 1`] = `ShallowWrapper {}`;
